src/MongoDB.Analyzer/Core/Builders/BuilderExpressionProcessor.cs

```diff
@@ -41,26 +41,21 @@ public static ExpressionsAnalysis ProcessSemanticModel(MongoAnalysisContext cont
 
         foreach (var node in root.DescendantNodesWithSkipList(nodesProcessed))
         {
-            SyntaxNode collectionNode = null;
+            SyntaxNode[] nodesToRewrite = null;
             var (nodeType, namedType, expressionNode) = GetNodeType(semanticModel, node);
 
             switch (nodeType)
             {
                 case NodeType.Builders:
                     {
+                        nodesToRewrite = GetBuildersDefinitionNodes(semanticModel, expressionNode).ToArray();
```

Comment:
So `ToArray` was moved to a different place. The goal was to avoid unnecessary copying of List to Array, in such transient processing.

Reply:
Done